### PR TITLE
diff: fix num is nil

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
fix issue https://internal.pingcap.net/jira/browse/TOOL-1289

### What is changed and how it works?
judge whether data is nil at the first when compare data

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: pkg/diff/merge.go [hunk 1]
```diff
@@ -47,35 +45,39 @@ func (r RowDatas) Less(i, j int) bool {
...
 				return false
 			}
 			return true
 		}
-		num1, err1 := strconv.ParseFloat(string(data1), 64)
+
+		num1, err1 := strconv.ParseFloat(strData1, 64)
 		if err1 != nil {
-			log.Fatal("convert string to float failed", zap.ByteString("data", data1), zap.Error(err1))
+			log.Fatal("convert string to float failed", zap.String("data", strData1), zap.Error(err1))
```
Comment (csuzhangxc) [inline pkg/diff/merge.go]: log the column name for debugging?

File: pkg/diff/merge.go [hunk 2]
```diff
@@ -47,35 +45,39 @@ func (r RowDatas) Less(i, j int) bool {
...
 			log.Fatal("data don't have column", zap.String("column", col.Name.O), zap.Reflect("data", r.Rows[j].Data))
 		}
-		data1 = col1.Data
-		data2 = col2.Data
 
-		if needQuotes(col.FieldType) {
-			strData1 := string(data1)
-			strData2 := string(data2)
+		strData1 := string(col1.Data)
+		strData2 := string(col2.Data)
```
Comment (csuzhangxc) [inline pkg/diff/merge.go]: move these two lines after L62.

File: pkg/diff/merge.go (hunk 1 above)
Comment (WangXiangUSTC) [reply pkg/diff/merge.go]: done https://github.com/pingcap/tidb-tools/pull/250/commits/d0ed8ba163809d2efa86718946636e59e3d1c5e7

File: pkg/diff/merge.go (hunk 2 above)
Comment (WangXiangUSTC) [reply pkg/diff/merge.go]: done https://github.com/pingcap/tidb-tools/pull/250/commits/d0ed8ba163809d2efa86718946636e59e3d1c5e7

**Review by csuzhangxc — COMMENTED** [review]
LGTM

File: pkg/diff/merge_test.go [hunk 3]
```diff
@@ -31,12 +30,12 @@ func (s *testMergerSuite) TestMerge(c *C) {
 	c.Assert(err, IsNil)
 
 	_, orderKeyCols := dbutil.SelectUniqueOrderKey(tableInfo)
-	ids := []int{3, 2, 2, 4, 1}
-	names := []string{"d", "b", "c", "b", "a"}
-	ages := []int{1, 2, 3, 4, 5}
+	ids := []string{"3", "2", "2", "4", "1", "NULL"}
+	names := []string{"d", "NULL", "c", "b", "a", "e"}
```
Comment (amyangfei) [inline pkg/diff/merge_test.go:34]: we can add more test cases to increase code path coverage later (easy to find out which code is not covered after added some coverage integration tool).

**Review by amyangfei — APPROVED** [review]
LGTM

File: pkg/diff/merge_test.go (hunk 3 above)
Comment (WangXiangUSTC) [reply pkg/diff/merge_test.go:34]: ok

**Pull request merged**